hr_timesheet_employee_required/tests/test_hr_timesheet_employee_required.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2018 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from psycopg2 import IntegrityError
+
+from odoo import fields
+from odoo.tests import common
+from odoo.tools.misc import mute_logger
+
+
+class TestHrTimesheetEmployeeRequired(common.TransactionCase):
+
+    def setUp(self):
+        super().setUp()
+
+        self.main_company = self.env.ref('base.main_company')
+        self.now = fields.Datetime.now()
+        self.User = self.env['res.users']
+        self.SudoUser = self.User.sudo()
+        self.Project = self.env['project.project']
+        self.SudoProject = self.Project.sudo()
+        self.HrEmployee = self.env['hr.employee']
+        self.SudoHrEmployee = self.HrEmployee.sudo()
+        self.AccountAnalyticLine = self.env['account.analytic.line']
+        self.SudoAccountAnalyticLine = self.AccountAnalyticLine.sudo()
+
+    def test_1(self):
+        project = self.SudoProject.create({
+            'name': 'Project #1',
+        })
+        user = self.SudoUser.create({
+            'name': 'User #1',
+            'login': 'test_hr_timesheet_employee_required_1',
+            'email': 'user-1@localhost',
+            'company_id': self.main_company.id,
+            'groups_id': [(6, 0, [
+                self.ref('hr_timesheet.group_hr_timesheet_user'),
```

Comment:
@alexey-pelykh It's `self.ref('hr_timesheet.group_hr_timesheet_user')` use multiple time so use as global variable.

Reply:
Done